addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -365,6 +369,12 @@ private void processPlayerStateChanged(String method, JsonObject json) {
             if (data.has("item")) {
                 JsonObject item = data.get("item").getAsJsonObject();
                 String mediaType = item.get("type").getAsString();
+                if (mediaType.equals("channel") && item.has("channeltype")) {
+                    String channelType = item.get("channeltype").getAsString();
+                    if (channelType.equals("radio")) {
+                        mediaType = "radio";
+                    }
+                }
                 listener.updateMediaType(mediaType);
```

Comment:
Just for information: I removed these lines in my local version and ran it since two weeks without observing any problems. The "mediatype" channel works like expected.